update Bloomberg Data and Analytics documentation and screenshots for… (#560)
MicrosoftDocs/powerquery-docs: powerquery-docs/connectors/bloomberg-data-and-analytics.md

---
title: Power Query Bloomberg Data and Analytics connector
description: Provides basic information and prerequisites for the Bloomberg Data and Analytics connector, descriptions of the optional input parameters, and discusses limitations and issues you might encounter.

> author: Shrewbitz

Reviewer: ```suggestion
author: DougKlopfenstein
```

Author: Does author mean something different than the person who wrote the documentation in this instance?

Reviewer: It usually matches the ms.author, but it doesn't have to. We can keep it as-is.

1. In Power BI Desktop, select **Home** > **Get Data**.

2. Select **Bloomberg Data and Analytics**, and then select **Connect**.

>    :::image type="content" source="./media/bloomberg-data-and-analytics/get-data.png" alt-text="Get data from Bloomberg Data and Analytics.":::

Reviewer: ```suggestion
   :::image type="content" source="./media/bloomberg-data-and-analytics/get-data.png" alt-text="Screenshot that shows get data from Bloomberg Data and Analytics.":::
```

4. Once you have written your query, select **OK** to execute it. The Bloomberg Data and Analytics connector now supports both **Bloomberg Query Language (BQL)** and **SQL** for data retrieval. This enhancement enables the connector to intelligently select the appropriate data source based on the query type you use.

   With the introduction of SQL support, you gain the benefit of [Query Folding](../query-folding-examples.md) in the Query Editor. This performance optimization, which reduces data transfer, is available exclusively for SQL queries.

   > [!NOTE]
   >The latest version of the connector introduces a unified query function, Bloomberg.Query(), in the formula bar. This function accepts both BQL and SQL, providing a streamlined experience. For existing reports developed with earlier versions, the BQL.Query() function remains operational and fully supported. Each function has its own authentication. To avoid logging in more than once, it is best practice to use Bloomberg.Query().
   :::image type="content" source="./media/bloomberg-data-and-analytics/function-bar.png" alt-text="Function bar queries.":::

>    **4a. Bloomberg Query Language (BQL) Example:**\

Reviewer: ```suggestion
   - **Bloomberg Query Language (BQL) Example:**
   
```

>     To learn more about BQL, contact your Bloomberg Sales representative.\

Reviewer: ```suggestion
    To learn more about BQL, contact your Bloomberg Sales representative.
    
```

>     :::image type="content" source="./media/bloomberg-data-and-analytics/bql-example.png" alt-text="Example BQL query.":::

Reviewer: ```suggestion
    :::image type="content" source="./media/bloomberg-data-and-analytics/bql-example.png" alt-text="Screenshot of an example BQL query.":::
```

>     **4b. SQL Query Example:**\

Reviewer: ```suggestion
    - **SQL Query Example:**
    
```

    :::image type="content" source="./media/bloomberg-data-and-analytics/sql-example.png" alt-text="Example SQL query.":::

5. To sign in to your Bloomberg account, select **Sign in**.

>    :::image type="content" source="./media/bloomberg-data-and-analytics/sign-in.png" alt-text="Select sign in button":::

Reviewer: ```suggestion
   :::image type="content" source="./media/bloomberg-data-and-analytics/sign-in.png" alt-text="Screenshot that shows the sign in button.":::
```

6. In the window that appears, provide your credentials to sign in to your Bloomberg account. Enter a terminal username and password, then select **Next**.

>     :::image type="content" source="./media/bloomberg-data-and-analytics/bb-sign-in.png" alt-text="Sign in to Bloomberg.":::

Reviewer: ```suggestion
    :::image type="content" source="./media/bloomberg-data-and-analytics/bb-sign-in.png" alt-text="Screenshot that shows how to sign in to Bloomberg.":::
```

7. Enter your Bloomberg B-Unit Mobile App code or B-Unit code and select **Submit**.

>     :::image type="content" source="./media/bloomberg-data-and-analytics/mobile-app.png" alt-text="Enter your auth code.":::

Reviewer: ```suggestion
    :::image type="content" source="./media/bloomberg-data-and-analytics/mobile-app.png" alt-text="Screenshot that shows where to enter your auth code.":::
```

8. Once you've successfully signed in, select **Connect**.

>    :::image type="content" source="./media/bloomberg-data-and-analytics/sign-in.png" alt-text="Select connect button.":::

Reviewer: ```suggestion
   :::image type="content" source="./media/bloomberg-data-and-analytics/sign-in.png" alt-text="Screenshot that shows the connect button.":::
```

9. After the connection is established, you will see data available for preview in the **Navigator**.

>    **9a. Bloomberg Query Language (BQL) Example:**\

Reviewer: ```suggestion
   - **Bloomberg Query Language (BQL) Example:**
   
```

>    :::image type="content" source="./media/bloomberg-data-and-analytics/bql-load.png" alt-text="Select BQL tables in the navigator view.":::\

Reviewer: ```suggestion
   :::image type="content" source="./media/bloomberg-data-and-analytics/bql-load.png" alt-text="Screenshot that shows where to select BQL tables in the navigator view.":::

```

>    **9b. SQL Query Example:**\

Reviewer: ```suggestion
   - **SQL Query Example:**
   
```

>    :::image type="content" source="./media/bloomberg-data-and-analytics/sql-load.png" alt-text="Select SQL tables in the navigator view.":::

Reviewer: ```suggestion
   :::image type="content" source="./media/bloomberg-data-and-analytics/sql-load.png" alt-text="Screenshot that shows where to select SQL tables in the navigator view.":::
```
